lib/install/management/appliance.go

```diff
@@ -219,6 +222,69 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		d.op.Warnf("Failed to remove datastore files for VM path %q: %s", folder, err)
 	}
 
+	// at this point the VCH should be gone and we need to cleanup the inventory
+	err = d.deleteVCHInventoryFolders()
+	if err != nil {
+		// if this fails manual cleanup is necessary
+		return err
+	}
+
+	return nil
+}
+
+func (d *Dispatcher) deleteVCHInventoryFolders() error {
+	parentFolderPath := path.Dir(d.appliance.InventoryPath)
+	// Now that the VCH is gone we will need to delete the inventory folder structure.
+	d.op.Debugf("Attempting to remove inventory folder: %s", parentFolderPath)
+
+	// grab the folder ref of the vm's direct parent
+	folderRef, err := d.session.Finder.Folder(d.op, parentFolderPath)
+	if err != nil {
+		d.op.Debugf("failed to find folder: %s", parentFolderPath)
+		return err
+	}
+
+	// we need to see if the folder is empty.
+	folderContents, err := folderRef.Children(d.op)
+	if err != nil {
+		return err
+	}
+
+	for len(folderContents) == 0 && folderRef.Reference() != d.session.VMFolder.Reference() {
+		// NOTE: Destroy on Inventory Folders is RECURSIVE, start from the leaf most target and check folder children to avoid undesired deletions.
+		err = d.removeFolder(folderRef)
```

Comment:
Don't delete here, keep track of the last one to delete. Since it is recursive. This will avoid needless vsphere calls.